bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/internal/GatewayCommand.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.openthermgateway.internal;
+
+import java.util.HashMap;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * @author Arjen Korevaar - Initial contribution
+ */
+@NonNullByDefault
+public class GatewayCommand {
+    private String code;
+    private String validationSet;
+    private String message;
+
+    public String getCode() {
+        return code;
+    }
+
+    public String getMessage() {
+        return this.message;
+    }
+
+    public String getValidationSet() {
+        return validationSet;
+    }
+
+    public String toFullString() {
+        return this.code + "=" + this.message;
+    }
+
+    private GatewayCommand(String code, String message, String validationSet) throws IllegalArgumentException {
+        this.code = code;
+        this.message = message;
+        this.validationSet = validationSet;
+
+        if (!validate()) {
+            throw new IllegalArgumentException(
+                    String.format("Invalid value '%s' for code '%s'", this.message, this.code));
+        }
+    }
+
+    private boolean validate() {
+        if (this.validationSet.isEmpty()) {
+            return true;
+        }
+
+        String[] validations = this.validationSet.split(",");
+
+        for (String validation : validations) {
+            if (this.message.equals(validation)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public static GatewayCommand parse(String code, String message) throws IllegalArgumentException {
+        if (code.isEmpty() && message.length() > 2 && message.charAt(2) == '=') {
+            return parse(message.substring(0, 2), message.substring(3));
+        }
+
+        if (code.length() == 2) {
+            String codeUpperCase = code.toUpperCase();
+
+            if (supportedCommands.containsKey(codeUpperCase)) {
+                String validateSet = supportedCommands.get(codeUpperCase);
+
+                if (validateSet == null) {
+                    validateSet = "";
+                }
+
+                return new GatewayCommand(codeUpperCase, message, validateSet);
+            } else {
+                throw new IllegalArgumentException(String.format("Unsupported gateway code '%s'", code.toUpperCase()));
+            }
+        }
+
+        throw new IllegalArgumentException(
+                String.format("Unable to parse gateway command with code '%s' and message '%s'", code, message));
+    }
+
+    private static final HashMap<String, @Nullable String> supportedCommands = getSupportedCommands();
```

Comment:
Please put static fields at the top of the class

Reply:
Moved to the top